shell/browser/native_window_mac.mm

```diff
@@ -1511,6 +1511,11 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
 
 void NativeWindowMac::SetWindowButtonVisibility(bool visible) {
   window_button_visibility_ = visible;
+  if (buttons_proxy_ && visible) {
+    [buttons_proxy_ redraw];
+    [buttons_proxy_ setVisible:YES];
```

Comment:
perf wart here, setVisible() will be called twice on 1516, 1522 if `visible && title_bar_style_ == kCustomButtonsOnHover`.

Maybe more importantly, it's kind of difficult to understand how the `title_bar_style_` and `buttons_proxy_` and `visible` are interacting in 1514-1524. This new paragraph is confusing to me when I read the comment  on 1519-1520 -- if we don't actually care about the style (the new paragraph doesn't check it) then why do we test for it on 1521?

Would this also be correct?

```objc++
void NativeWindowMac::SetWindowButtonVisibility(bool visible) {
  window_button_visibility_ = visible;
  // The visibility of window buttons are managed by |buttons_proxy_| if the
  // style is customButtonsOnHover.
  if (title_bar_style_ == TitleBarStyle::kCustomButtonsOnHover) {
    if (visible)
      [buttons_proxy_ redraw];
    [buttons_proxy_ setVisible:visible];
  } else {
    InternalSetWindowButtonVisibility(visible);
  } 
  NotifyLayoutWindowControlsOverlay();
}   
```

Reply:
@ckerr this issue is not specific to `TitleBarStyle::kCustomButtonsOnHover` - the buttons  need to be redrawn if `buttons_proxy_` is being used for all titlebar styles which make use of it

Reply:
Just pushed up a change - think that should do it though i'd like to make sure i didn't miss anything cc @zcbenz who added this initially!

Reply:
hmm @ckerr looks like this did in fact kill it